addons/account/models/account_account.py

```diff
@@ -253,9 +275,13 @@ def _check_journal_consistency(self):
                 account=account.display_name
             ))
 
-    @api.constrains('company_id')
+    @api.constrains('company_ids')
     def _check_company_consistency(self):
```

Comment:
This will prevent adding companies to an account if the account is already used anywhere

Reply:
That's true. Instead, we just want to prevent removing companies from the account if there are already account move lines created for that company and account.